PalasoUIWindowsForms/ClearShare/CreativeCommonsLicense.cs

```diff
@@ -193,32 +194,73 @@ public override string Token
+				s += GetComponentOfLicenseInBestLanguage("CommercialUseAllowed", "You are free to make commercial use of this work.", languagePriorityIds, out idOfLanguageUsed) + " ";
 			else
-				s += "You may not use this work for commercial purposes. ";
+				s += GetComponentOfLicenseInBestLanguage("NonCommercial", "You may not use this work for commercial purposes.", languagePriorityIds, out idOfLanguageUsed) + " ";
 
 			if(DerivativeRule == DerivativeRules.Derivatives)
-				s += "You are free to adapt, remix, copy, distribute, and transmit this work. ";
+				s += GetComponentOfLicenseInBestLanguage("Derivatives", "You may adapt and add to this work.", languagePriorityIds, out idOfLanguageUsed) + " ";
 
 			if (DerivativeRule == DerivativeRules.NoDerivatives)
-				s += "You may not alter, transform, or build upon this work without permission. ";
+				s += GetComponentOfLicenseInBestLanguage("NoDerivatives", "You may not make changes or build upon this work without permission.", languagePriorityIds, out idOfLanguageUsed) + " ";
 
 			if (DerivativeRule == DerivativeRules.DerivativesWithShareAndShareAlike)
-				s += "You may adapt or build upon this work, but you may distribute the resulting work only under the same or similar license to this one.";
+				s += GetComponentOfLicenseInBestLanguage("DerivativesWithShareAndShareAlike", "You may adapt and add to this work, but you may distribute the resulting work only under the same or similar license to this one.", languagePriorityIds, out idOfLanguageUsed) + " ";
 
 			if (AttributionRequired)
-				s += "You must attribute the work in the manner specified by the author. ";
+				s += GetComponentOfLicenseInBestLanguage("AttributionRequired", "You must keep the copyright and credits for authors, illustrators, etc.", languagePriorityIds, out idOfLanguageUsed) + " ";
 
-			return s;
+			return s.Trim();
+		}
+
+		private string GetComponentOfLicenseInBestLanguage(string idSuffix, string englishText, IEnumerable<string> languagePriorityIds, out string idOfLanguageUsed)
+		{
+			var id = "MetadataDisplay.Licenses.CreativeCommons." + idSuffix;
+			var comment = "See http://creativecommons.org/ to find out how this is normally translated in your language. What we're aiming for here is an easy to understand summary.";
+			foreach(var targetLanguage in languagePriorityIds)
+			{
+				if (targetLanguage == "en")
+				{
+					//do the query to make sure the string is there to be translated someday
+					var unused = LocalizationManager.GetDynamicString("Palaso", id, englishText, comment);
+					idOfLanguageUsed = "en";
+					return englishText;
+				}
+				//otherwise, see if we have a translation
+				if (LocalizationManager.GetIsStringAvailableForLangId(id, targetLanguage))
+				{
+					idOfLanguageUsed = targetLanguage;
+					return LocalizationManager.GetDynamicStringOrEnglish("Palaso", id, englishText, comment, targetLanguage);
+				}
+			}
+			idOfLanguageUsed = string.Empty;
```

Comment:
Yes, that's really up to the client to deal with. Since the client already has to come up with a prioritized list of languages, it seems like it complicates this API to add "and by the way, we always add English".